### PR TITLE
Add new listeners

### PR DESCRIPTION
**Pull request by Half-Shot**
Made changes according to #46

### CODE REVIEW

File: matrix_client/client.py [hunk 1]
```diff
@@ -238,7 +244,7 @@ def listen_forever(self, timeout_ms=30000):
                 else:
                     raise e
             except Exception as e:
-                self.logger.error("Exception thrown during sync\n %s", str(e))
+                self.logger.error("Exception thrown during sync\n %s", e)
```
Comment (erikjohnston) [inline matrix_client/client.py]: this should probably just use:

```python
self.logger.exception("Exception thrown during sync")
```

Comment (erikjohnston) [inline matrix_client/client.py]: Also, why is the logger not defined at the module level as:

```python
logger = logging.getLogger(__name__)
```

**Comment by erikjohnston** [comment]
flaaaaaaaaaaake8

**Comment by Half-Shot** [comment]
...I fixed and amended this...git please 😭

**Pull request merged**